base/linalg/generic.jl

```diff
@@ -977,29 +977,36 @@ julia> a = [1 2; 2 -1]
 julia> istriu(a)
 false
 
+julia> istriu(a, -1)
+true
+
 julia> b = [1 im; 0 -1]
 2×2 Array{Complex{Int64},2}:
  1+0im   0+1im
  0+0im  -1+0im
 
 julia> istriu(b)
 true
+
+julia> istriu(b, 1)
+false
 ```
 """
-function istriu(A::AbstractMatrix)
+function istriu(A::AbstractMatrix, k::Integer = 0)
     m, n = size(A)
-    for j = 1:min(n,m-1), i = j+1:m
-        if !iszero(A[i,j])
-            return false
+    for j in 1:min(n, m + k - 1)
+        for i in max(1, j - k + 1):m
+            iszero(A[i, j]) || return false
```

Comment:
maybe `@inbounds`?

Reply:
I am not certain this is safe for arbitrary `AbstractArray`s, for example `OffsetArray`s, and so hesitate to add an `@inbounds`?

Reply:
Eventually, these functions should use either be restricted to `StridedMatrix` or the index calculations be changed to work with `OffsetArrays`.

Reply:
Good call :). Do you have thoughts on the best way to generalize these loops? Thanks!

Reply:
The easiest might be to reuse the existing loops and then use the `i` and `j` to index into the actual indices for the array, e.g. `ii = indices(A, 1)[i]`, and then index with `ii` and `jj`.